src/core/import-creds.c

```diff
@@ -713,5 +713,15 @@ int import_credentials(void) {
                         r = q;
         }
 
+        if (r >= 0) {
+                _cleanup_free_ char *address = NULL;
+
+                r = read_credential("notify_socket", (void **)&address, /* ret_size= */ NULL);
+                if (r < 0 && r != -ENOENT)
```

Comment:
also check for ENXIO (i.e. ENOENT: creds passed but the requested one is missing. ENXIO: env var not set, hence no creds passed)